create auto rng reporter

tkernell opened the pull request:
- created new reporter type RNGReporter
- submits a new random number in 30 minute intervals
- checks every 5 minutes whether a new interval can be submitted

oraclown commented on src/telliot_feed_examples/reporters/rng_interval.py:
```diff
@@ -0,0 +1,355 @@
+        token: Contract,
+        autopay: Contract,
+        stake: float = 10.0,
+        datafeed: Optional[DataFeed[Any]] = None,
```
I think you want to inherit from `PolygonReporter` instead of `IntervalReporter`, no? Also, leave out everything from the constructor except for overwriting the datafeed with an RNG datafeed like how you're doing in `report_once()`:
```python
self.datafeed = await assemble_rng_datafeed(
            timestamp=rng_timestamp, node=self.endpoint, account=self.account
        )
```

You also don't have to include any of the class methods that you aren't changing (like `ensure_staked()` or `check_reporter_lock()`), since they're inherited.

oraclown reviewed (changes requested):
Nice start. I left some feedback. Also add a test to submit a value using the RNGReporter maybe.

tkernell commented:
- `RNGReporter` now inherits from `PolygonReporter`
- removed the local constructor
- removed all unchanged class methods
- added a test for submitting with `RNGReporter`

oraclown reviewed (approved):
Nice changes. Going to merge this. Be prepared to come back and handle errors Spuddy might find. Also, you might want to update the docs on using the `rng_auto` flag.
